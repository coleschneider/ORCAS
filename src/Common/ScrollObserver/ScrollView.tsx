import * as React from 'react';
import ScrollIntoView, { ScrollHelper, FooterScrollHelper } from 'Common/ScrollObserver/ScrollIntoView/ScrollIntoView';
import { compose, defaultProps, withHandlers } from 'recompose';

interface DefaultProps {
  initial?: string;
  Text?: string;
}

interface EnhanceProps {
  scrollIntoView: (ev: React.MouseEvent) => void;
}
interface OuterProps {
  activeElement: string;
  cb: any;
  selector: string;
  Text: string;
}

export const withScrollView = compose(
  defaultProps<DefaultProps>({ initial: 'Home', Text: 'Home' }),
  withHandlers<OuterProps, EnhanceProps>({
    scrollIntoView: ({ selector, cb }) => () => {
      const el = document.querySelector(selector);
      if (typeof cb === 'function') {
        cb();
      }
      el.scrollIntoView({ behavior: 'smooth' });
    },
  }),
);
export const ViewScrollHelper = withScrollView(ScrollHelper);
export const FooterViewScroll = withScrollView(FooterScrollHelper);
export default withScrollView(ScrollIntoView);
