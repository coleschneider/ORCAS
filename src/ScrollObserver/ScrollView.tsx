import * as React from 'react';
import ScrollIntoView from './ScrollIntoView/ScrollIntoView';
import { compose, defaultProps, withHandlers } from 'recompose';

interface DefaultProps {
  initial?: string;
  Text?: string;
}

interface EnhanceProps {
  scrollIntoView: (ev: React.MouseEvent) => void;
}
interface OuterProps {
  activeElement: string;
  cb: any;
  selector: string;
  Text: string;
}

export const withScrollView = compose(
  defaultProps<DefaultProps>({ initial: 'Home', Text: 'Home' }),
  withHandlers<OuterProps, EnhanceProps>({
    scrollIntoView: ({ selector, cb }) => () => {
      const el = document.querySelector(selector);
      if(typeof cb === 'function'){
        cb()
      }
      el.scrollIntoView({ behavior: 'smooth' });
    },
  }),
);

export default withScrollView(ScrollIntoView);
