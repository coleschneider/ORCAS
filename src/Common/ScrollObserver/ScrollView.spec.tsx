import * as React from 'react';
import ScrollView, { withScrollView } from './ScrollView';
import { shallow, mount } from 'enzyme';

describe('ScrollView', () => {
  describe('withScrollView', () => {
    it('sets the initial props of the composed function', () => {
      const WrapperHoc = withScrollView(jest.fn());
      const wrapper = shallow(<WrapperHoc />);
      expect(wrapper.props().Text).toBe('Home');
      expect(wrapper.props().initial).toBe('Home');
    });
    it('sets initial to home when props is undefeined', () => {
      const WrapperHoc = withScrollView(jest.fn());
      const wrapper = shallow(<WrapperHoc />);
      expect(wrapper.props().Text).toBe('Home');
      expect(wrapper.props().initial).toBe('Home');
    });
  });

  describe('ScrollIntoView', () => {
    const docSpy = jest.spyOn(document, 'querySelector');
    const elSpy = jest.fn().mockImplementation(() => {
      props.activeElement = 'NEXT LINK';
    });
    const props = {
      Text: 'test',
      activeElement: 'test',
      selector: '#test',
      setActiveElement: elSpy,
    };

    it('matches the snapshot', () => {
      const wrapper = shallow(<ScrollView {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should find the element and call the method with smooth', () => {
      const node = document.querySelector('#scroll-test');
      node.scrollIntoView = jest.fn();
      const wrapper = mount(<ScrollView selector="#scroll-test" />);
      wrapper.find('ScrollIntoView').simulate('click');
      expect(node.scrollIntoView.mock.calls[0][0]).toMatchObject({
        behavior: 'smooth',
      });
    });
  });
});
