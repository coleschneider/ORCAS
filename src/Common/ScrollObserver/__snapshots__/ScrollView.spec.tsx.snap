// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollView ScrollIntoView matches the snapshot 1`] = `
<withHandlers(ScrollIntoView)
  Text="test"
  activeElement="test"
  initial="Home"
  selector="#test"
  setActiveElement={[MockFunction]}
/>
`;
